flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/batch/sql/TableSourceTest.xml

```diff
@@ -109,13 +109,13 @@ FROM T
       <![CDATA[
 LogicalProject(id=[$0], nested1=[$1.nested1], results=[+(+($1.nested1.value, $1.nested2.num), $2)])
 +- LogicalProject(id=[$0], deepNested=[$1], metadata_1=[$2], metadata_2=[$3])
-   +- LogicalTableScan(table=[[default_catalog, default_database, T]])
+   +- LogicalTableScan(table=[[default_catalog, default_database, T, metadata=[metadata_1, metadata_2]]])
 ]]>
     </Resource>
     <Resource name="optimized exec plan">
       <![CDATA[
 Calc(select=[id, deepNested_nested1 AS nested1, ((deepNested_nested1.value + deepNested_nested2_num) + metadata_1) AS results])
-+- TableSourceScan(table=[[default_catalog, default_database, T, project=[id, deepNested_nested1, deepNested_nested2_num], metadata=[metadata_1]]], fields=[id, deepNested_nested1, deepNested_nested2_num, metadata_1])
++- TableSourceScan(table=[[default_catalog, default_database, T, metadata=[metadata_1, metadata_2], project=[id, deepNested_nested1, deepNested_nested2_num], metadata=[metadata_1]]], fields=[id, deepNested_nested1, deepNested_nested2_num, metadata_1])
```

Comment:
This looks wrong. The source should use only `metadata_1`. Looking at path revisions of this class, this was the case last year.

Reply:
I just noticed that the parenthesis are also not correct. What is going on here?

Reply:
> I just noticed that the parenthesis are also not correct. What is going on here?

Where are they wrong?

> This looks wrong. The source should use only metadata_1. Looking at path revisions of this class, this was the case last year.

Bear in mind there are two `ReadingMetadataSpec`. One applied during a scan and one during projection pushdown. That's also the case when creating the `ExecNode` tree that `applyReadableMetadata` is called twice.
The spec from projection pushdown uses only `metadata_1`.

Reply:
I updated the PR so that we leave only a single instance of each spec